editorModule.js

```diff
@@ -31,30 +32,272 @@ export default class EditorModule{
+            autocompleteJSON = await response.json()
+        } catch(e) {
+            console.error("Error fetching autocomplete suggestions!", e);
+        }
+
+        async function ShowAutocompletion(obj) { 
+            // Disable default autocompletion for javascript
+            monaco.languages.typescript.javascriptDefaults.setCompilerOptions({ noLib: true, allowNonTsExtensions: true });
+            monaco.languages.typescript.javascriptDefaults.setEagerModelSync(true);
+          
+            // Helper function to return the monaco completion item type of a thing
+            function getType(typeName, isMember) {
+                isMember =  (isMember == undefined) ? (typeof isMember == "boolean") ? isMember : false : false; // Give isMember a default value of false
+
+                switch (typeName) { 
+                    case "object": 
+                        return monaco.languages.CompletionItemKind.Class;
+
+                    case "function": 
+                        return (isMember) ? monaco.languages.CompletionItemKind.Method : monaco.languages.CompletionItemKind.Function;
+
+                    default: 
+                        return (isMember) ? monaco.languages.CompletionItemKind.Property : monaco.languages.CompletionItemKind.Variable;
+                }
+            }
+
+            function getActiveWord(model, position) {
+                // Split everything the user has typed on the current line up at each space, and only look at the last word
+                var last_chars = model.getValueInRange({startLineNumber: position.lineNumber, startColumn: 0, endLineNumber: position.lineNumber, endColumn: position.column});
+                // Clean up any white space
+                while(last_chars.indexOf("  ") >= 0 || last_chars.indexOf("\t") >= 0) {
+                    last_chars = last_chars.replace(/\t/g, " ").replace(/\s\s/g, " ")
+                }
+                // What the user is currently typing (everything after the last space)
+                var words = last_chars.split(";");
+                return words[words.length - 1].trim();
+            }
+
+            // Register object that will return autocomplete items 
+            monaco.languages.registerCompletionItemProvider('*', {
+                // Run this function when the period or space is typed
+                triggerCharacters: ['.', ' '],
+
+                // Function to generate autocompletion results
+                provideCompletionItems: function(model, position, token) {
+
+                    // Return if this is not the javascript editor
+                    if(model.id != '$model3') {
+                        return
+                    }
 
-        //27, javascript
-        const javascriptLoader = await monaco.languages.getLanguages()[27].loader();
+                    // Get current word being typed
+                    var active_typing = getActiveWord(model, position);
```

Comment:
Prefer const/let. Which was already being used. Lots of instances to fix in this PR.